### ISSUE TITLE
Feature Request : Blender styled navigation

### ISSUE THREAD

**Issue by therealrobster** [guillaumechereau/goxel#192]
Hi there,

Loving the software, just found it.   Nice work, will update works as I make them.

For now though, it seems the open source world has adopted the Blender navigation system.  Blender, FreeCAD, Godot and many other 3D systems have this option.  It would be great if Goxel did also so we can move between our tools and just have it work.

Just a suggestion.

Thanks for Goxel

**Comment by guillaumechereau** [comment]
Hello, what do you mean by blender navigation system exactly?

**Comment by therealrobster** [comment]
Hey there!  Thanks for the interest.

Blender navigation is a specific set of key/mouse combination that all major open source apps have been adopting.  Users can move between the apps effortlessly this way.

https://docs.blender.org/manual/en/latest/editors/3dview/navigate/navigation.html  That is the official doc but it's a bit dry.  It does show specifics though.

Essentially you can:
Orbit : MMB
Pan : Shift MMB
Zoom : CTRL MMB (as well as scroll wheel)

The link shows the rest.

Although that may seem like an extra step to your key system (middle and right mouse only) if it's drummed into your brain like it is for so many people, it's really difficult to learn another system when moving to goxel. I keep muffing it this morning trying.  I'll get there no doubt. I love your software, but it'd be great to be able to keep consistency between all my apps that way.

I hope that helps.  Thanks again

**Comment by guillaumechereau** [comment]
Thanks for the link.  Yes I think it would make sense to use the same navigation as blender.  Maybe with some options to change it if needed.

**Comment by therealrobster** [comment]
Definitely.  All the apps seem to have options for their own bindings, Blender, Maya and sometimes others.  Blender is the big one in the open source world though so it makes sense.

**Comment by YBPN** [comment]
It would be cool to have a version of this for the ipad as well:

Two-finger swipe: pan (already a feature)
Two-finger pinch: zoom (already a feature)
Three-finger swipe: orbit (including pitch)
If using apple pencil: one-finger orbit (including pitch)

This would make the navigation experience smoother, so that the scrollbar at the side can be just optional.

**Comment by guillaumechereau** [comment]
I just made some change to support control and shift key with the middle mouse button.  Haven't tested it on mac yet.

**Comment by asciifaceman** [comment]
The biggest issue right now is basically being unable to use goxel on a laptop with a buttonless touch pad - blender style navigation has emulation with alt/ctl/shift keys to allow for panning etc without the mouse buttons for touchpad support

**Comment by guillaumechereau** [comment]
Right, this should be implemented.
